domain.go

```diff
@@ -1107,11 +1107,11 @@ func (d *Domain) GetState() (DomainState, int, error) {
 // See also https://libvirt.org/html/libvirt-libvirt-domain.html#virDomainGetID
 func (d *Domain) GetID() (uint, error) {
 	var err C.virError
-	id := uint(C.virDomainGetIDWrapper(d.ptr, &err))
-	if id == ^uint(0) {
-		return id, makeError(&err)
+	id := C.virDomainGetIDWrapper(d.ptr, &err)
+	if id == ^C.uint(0) {
+		return uint(id), makeError(&err)
 	}
-	return id, nil
+	return uint(id), nil
```

Comment:
I'm not seeing the point / benefit of this change ? AFAICT it is functionally identical to the existing code. What am I missing ?

Reply:
For me the following code yields `^uint(0): 18446744073709551615 ^C.uint(0): 4294967295`.

```
package main

import "C"

import (
	"fmt"
)

func main() {
	fmt.Printf("^uint(0): %d ^C.uint(0): %d\n", ^uint(0), ^C.uint(0))
}
```

Reply:
On compilers where a C.uint is 32 bits, when virDomainGetIDWrapper returns a C int of -1, id is set to be a Go uint of value 2^32-1, as opposed to 2^64-1. The original code assumes that Go's uint is the same size as C's uint.

Reply:
Ah yes, thanks for pointing that out.